phoebe/backend/contacts_smoothing.py

```diff
@@ -0,0 +1,241 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from itertools import combinations
+
+
+def _isolate_neck(coords_all, teffs_all, cutoff = 0.,component=1, plot=False):
+    if component == 1:
+        cond = coords_all[:,0] >= 0+cutoff
+    elif component == 2:
+        cond = coords_all[:,0] <= 1-cutoff
+    else:
+        raise ValueError
+        
+    if plot:
+        fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(10,5))
+        axes[0].scatter(coords_all[cond][:,0], coords_all[cond][:,1])
+        axes[1].scatter(coords_all[cond][:,0], teffs_all[cond])
+        axes[0].set_xlabel('x')
+        axes[0].set_ylabel('y')
+        axes[1].set_xlabel('x')
+        axes[1].set_ylabel('teff')
+        fig.tight_layout()
+        plt.show()
+     
+    return coords_all[cond], teffs_all[cond], np.argwhere(cond).flatten()
+
+
+def _dist(p1, p2):
```

Comment:
can we not use np.dot() or similar here?

Reply:
good point! (no pun intended)